src/charm.py

```diff
@@ -361,6 +361,9 @@ def join_unit_to_cluster(self) -> None:
                 # Stop GR for cases where the instance was previously part of the cluster
                 # harmless otherwise
                 self._mysql.stop_group_replication()
+                self._mysql.rescan_cluster_if_instance_already_in_cluster(
```

Comment:
In addition to [comment on the lib](https://github.com/canonical/mysql-operator/pull/462#discussion_r1638086525) :

```suggestion
                if self._mysql.is_instance_in_cluster(...
                  # ensure joining instance is not present in stale cluster metadata
                  self._mysql.rescan(..
```

Reply:
addressed in [ecd1150](https://github.com/canonical/mysql-k8s-operator/pull/433/commits/ecd1150121fdf62bcb2f87f86d5eda7a43e775b7)